sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/SparkThriftServerProtocolVersionsSuite.scala

```diff
@@ -458,5 +459,17 @@ class SparkThriftServerProtocolVersionsSuite extends HiveThriftServer2TestBase {
         }
       }
     }
+
+    test(s"SPARK-35017: $version get day-time interval type") {
```

Comment:
I tested only day-time interval because constructing of year-month interval in SQL is not possible at the moment. I created the JIRA for that: SPARK-35018

Reply:
The PR https://github.com/apache/spark/pull/32240 adds a test for year-month intervals.